sdk/python/kfp/client/client.py

```diff
@@ -689,9 +689,12 @@ def _choose_pipeline_file(file_list: List[str]) -> str:
                 f'The package_file {package_file} should end with one of the '
                 'following formats: [.tar.gz, .tgz, .zip, .yaml, .yml].')
 
-    def _override_caching_options(self, workflow: str,
+    def _override_caching_options(self, workflow: dict, 
                                   enable_caching: bool) -> None:
-        raise NotImplementedError('enable_caching is not supported yet.')
+        if workflow is not None:
+            for task in workflow['root']['dag']['tasks']:
+                if 'cachingOptions' in workflow['root']['dag']['tasks'][task]:
```

Comment:
There is one case where this would not have intended behavior:

This currently works in every case where the IR is compiled by the Python SDK, because the `cachingOptions` message is always explicitly written by the SDK.

However, this isn't a strict requirement for IR. For example, if IR is hand-written a user could omit the `cachingOptions` field and fall back to [default](https://github.com/kubeflow/pipelines/blob/b0a7ba5ef408f44ee415f5863dac52f6cae105f9/api/v2alpha1/pipeline_spec.proto#L425-L429) (`enableCache: false`). If a user wanted to override with `True` in this case, `if 'cachingOptions' in workflow` would be `False`, the override would not change the IR, and the IR would fall back to the default case: `False`.